libs/nx-terraform/src/generators/create-environment-sp/generator.ts

```diff
@@ -0,0 +1,97 @@
+import { Tree } from '@nrwl/devkit'
+import execa from 'execa'
+import { isDryRun } from '../../common/isDryRun'
+import { readRepoSettings } from '../../common/read-repo-settings'
+import { readConfigFromEnvFile } from '../../common/readConfigFromEnvFile'
+import { NxTerraformCreateEnvironmentServicePrincipalSchema } from './schema'
+
+export default async function (
+    tree: Tree,
+    options: NxTerraformCreateEnvironmentServicePrincipalSchema,
+) {
+    const repoSettings = readRepoSettings()
+    const environmentConfig = await readConfigFromEnvFile(
+        repoSettings.terraformStateType,
+        options.environmentName,
+    )
+
+    const servicePrincipalName = `gh-actions-${repoSettings.azureResourcePrefix}-${options.environmentName}-sp`
+    const scopes = `/subscriptions/${environmentConfig.subscriptionId}/resourcegroups/${environmentConfig.resourceGroupName}`
+    const containerScope = `/subscriptions/${environmentConfig.subscriptionId}/resourceGroups/${environmentConfig.resourceGroupName}/providers/Microsoft.Storage/storageAccounts/${environmentConfig.terraformStorageAccount}/blobServices/default/containers/${environmentConfig.terraformStorageContainer}`
+
+    if (isDryRun()) {
+        console.log('Will run:')
+
+        console.log(`> az ad sp create-for-rbac --role ${options.role} \
+    --scopes ${scopes} \
+    --name ${servicePrincipalName} \
+    --sdk-auth`)
+
+        console.log(`> az role assignment create \
```

Comment:
Do we need to cater to not using storage for TF state?

Reply:
I.e. this wouldn't exist in that case

Reply:
Fixed